test/infrastructure/docker/internal/provisioning/cloudinit/writefiles.go

```diff
@@ -62,13 +66,79 @@ func newWriteFilesAction() action {
 	return &writeFilesAction{}
 }
 
-func (a *writeFilesAction) Unmarshal(userData []byte) error {
+func (a *writeFilesAction) Unmarshal(userData []byte, kindMapping kind.Mapping) error {
 	if err := yaml.Unmarshal(userData, a); err != nil {
 		return errors.Wrapf(err, "error parsing write_files action: %s", userData)
 	}
+	for i, f := range a.Files {
+		if f.Path == kubeadmInitPath {
+			// NOTE: in case of join the kubeadmConfigFile contains both the ClusterConfiguration and the InitConfiguration
+			contentSplit := strings.Split(f.Content, "---\n")
+
+			if len(contentSplit) != 3 {
```

Comment:
Is there a way with our regular APIs that folks add more to this file? (e.g. like KubeletConfiguration)

Just wondering if we should gracefully handle this and just try to parse all elements of the contentSplit slice

Reply:
It is hardcoded in CAPBK, users can't change it